src/server/scripts/EasternKingdoms/zone_silverpine_forest.cpp

```diff
@@ -941,10 +943,488 @@ class spell_silverpine_forsaken_trooper_masterscript_high_command : public Spell
+
+    void Reset() override
+    {
+        _events.Reset();
+
+        // Note: these texts are not sent by summoned creatures from Lordaeron (questId 27098).
+        if (!me->IsSummon())
+        {
+            if (urand(0, 1))
+                Talk(TALK_TROOPER_RESET);
+        }
+    }
+
+    void DamageTaken(Unit* attacker, uint32& damage, DamageEffectType /*damageType*/, SpellInfo const* /*spellInfo = nullptr*/) override
+    {
+        // HACKFIX: sparring system is not implemented yet.
+        if (!attacker->IsPlayer() && me->HealthBelowPctDamaged(85.0f, damage))
+            damage = 0;
+    }
+
+    void JustEngagedWith(Unit* /*who*/) override
+    {
+        _events.ScheduleEvent(EVENT_CLEAVE, 5s, 8s);
+    }
+
+    void UpdateAI(uint32 diff) override
+    {
+        _events.Update(diff);
+
+        while (uint32 eventId = _events.ExecuteEvent())
+        {
+            switch (eventId)
+            {
+                case EVENT_CLEAVE:
+                    DoCastVictim(SPELL_CLEAVE);
+                    _events.Repeat(5s, 8s);
+                    break;
+
+                default:
+                    break;
+            }
+        }
+
+        if (!UpdateVictim())
+            return;
+
+        DoMeleeAttackIfReady();
+    }
+
+private:
+    EventMap _events;
+};
+
+enum BatHandlerMaggothbreath
+{
+    QUEST_ITERATING_UPON_SUCCESS                = 26998,
+
+    SPELL_SUMMON_FORSAKEN_BAT                   = 83584,
+
+    TALK_MAGGOTHBREATH_ITERATING_UPON_SUCCESS   = 0,
+
+    DATA_GOSSIP_MENU_MAGGOT                     = 11892
+};
+
+// Bat Handler Maggotbreath - 44825
+struct npc_silverpine_bat_handler_maggotbreath : public ScriptedAI
+{
+    npc_silverpine_bat_handler_maggotbreath(Creature* creature) : ScriptedAI(creature) { }
+
+    bool OnGossipHello(Player* player) override
+    {
+        if (player->GetQuestStatus(QUEST_ITERATING_UPON_SUCCESS) == QUEST_STATUS_INCOMPLETE)
+            AddGossipItemFor(player, DATA_GOSSIP_MENU_MAGGOT, 0, GOSSIP_SENDER_MAIN, GOSSIP_ACTION_INFO_DEF);
+
+        if (me->IsTaxi())
+            AddGossipItemFor(player, DATA_GOSSIP_MENU_MAGGOT, 1, GOSSIP_SENDER_MAIN, GOSSIP_ACTION_INFO_DEF + 1);
+
+        SendGossipMenuFor(player, player->GetGossipTextId(me), me->GetGUID());
+        return true;
+    }
+
+    bool OnGossipSelect(Player* player, uint32 /*menuId*/, uint32 gossipListId) override
+    {
+        switch (player->PlayerTalkClass->GetGossipOptionAction(gossipListId))
+        {
+            case GOSSIP_ACTION_INFO_DEF:
+                Talk(TALK_MAGGOTHBREATH_ITERATING_UPON_SUCCESS, player);
+                player->CastSpell(player, SPELL_SUMMON_FORSAKEN_BAT, true);
+                break;
+
+            // Note: Blizzard messed up this OptionNpc. It sends GossipOptionNpc::None rather than GossipOptionNpc::TaxiNode,
+            // making it useless. To keep it blizzlike, we're just sending TaxiMenu forcefully as well.
```

Comment:
This is weird

Reply:
I know, I was also baffled by this. I had to ask @meji46 as well for this, but the SQL data comes from their data directly.